packages/vue/src/components/molecules/SfModal/SfModal.js

```diff
@@ -48,48 +48,24 @@ export default {
   },
   methods: {
     close() {
-      this.$emit("close");
-      // remove left code
-    },
-    keydownHandler(e) {
-      if (e.key === "Escape" || e.key === "Esc" || e.keyCode === 27) {
-        this.close();
-      }
-      if (this.visible) {
-        // TODO: this is an error right now. what shall it do?
-        //  document.addEventListener("keydown", keydownHandler);
-        this.$emit("close");
-      } else {
-        // TODO: this is an error right now. what shall it do?
-        //  document.removeEventListener("keydown", keydownHandler);
-        this.$emit("open");
-      }
+      this.$emit("input", !this.value);
```

Comment:
I really don't think `input` is a good event here :grimacing:
I think you've set it just to support `v-model`, right? We can fix it by adding custom model event, as I've suggested in previous comment

```suggestion
      this.$emit("toggle", !this.value);
```

or

```suggestion
      this.$emit("update:visible", !this.value);
```

Reply:
https://github.com/DivanteLtd/storefront-ui/pull/459